### PR TITLE
Adding gVisor Integration tests 

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:
Adding integration tests for the gVisor container runtime integration in Gardener.

To test for the first release of [gVisor extension](https://github.com/gardener/gardener-extension-runtime-gvisor) .

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Has the same suboptimal behaviour as discussed in this [PR](https://github.com/gardener/gardener/pull/2201#discussion_r411907231) - is restricted to Ubuntu OS. 
I think we can still go ahead with this PR as the already existing containerd integration test is configured the same way. As soon as there is a solution, these two integration tests can be easily adjusted. 

**Release note**:

```improvement operator

```

### CODE REVIEW

File: test/integration/shoots/operations/containerruntime.go [hunk 1]
```diff
@@ -107,19 +86,240 @@ var _ = ginkgo.Describe("Shoot container runtime testing", func() {
...
 	response, err := rootPodExecutor.Execute(ctx, command)
 	framework.ExpectNoError(err)
 	g.Expect(response).ToNot(g.BeNil())
 	g.Expect(string(response)).To(g.Equal(fmt.Sprintf("%s\n", expected)))
 }
+
+// deployGVisorPod deploys a pod using the gVisor RuntimeClass.
+func deployGVisorPod(ctx context.Context, c client.Client) (*corev1.Pod, error) {
+	allowedCharacters := "0123456789abcdefghijklmnopqrstuvwxyz"
+	id, err := gardenerutils.GenerateRandomStringFromCharset(3, allowedCharacters)
```
Comment (ialidzhikov) [inline test/integration/shoots/operations/containerruntime.go:291]: You could use .metadata.generateName for the Pod name.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: test/integration/shoots/operations/containerruntime.go (hunk 1 above)
Comment (danielfoehrKn) [reply test/integration/shoots/operations/containerruntime.go:291]: true, then I ll also adjust that in the test-framework where it is used similarely.

File: pkg/apis/core/types_shoot.go [hunk 2]
```diff
@@ -723,6 +723,13 @@ const (
 	CRINameContainerD CRIName = "containerd"
 )
 
+// ContainerRuntimeName is a type alias for the ContainerRuntime name string.
+type ContainerRuntimeName string
+
+const (
+	ContainerRuntimeGVisor ContainerRuntimeName = "gvisor"
```
Comment (timebertt) [inline pkg/apis/core/types_shoot.go:730]: Why is this here?
I don't think, there should be any known provider names in Gardener's API...
Btw, isn't this a clear case for an integration test, that should be part of the respective Extension repo?

**Review by dguendisch — APPROVED** [review]
integrationtest-wise lgtm

File: test/integration/shoots/operations/containerruntime.go [hunk 3]
```diff
@@ -107,19 +86,240 @@ var _ = ginkgo.Describe("Shoot container runtime testing", func() {
...
+				{
+					Name:  "gvisor-container",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"10000000",
+					},
+					Resources:                corev1.ResourceRequirements{},
+					TerminationMessagePath:   "/dev/termination-log",
+					TerminationMessagePolicy: corev1.TerminationMessageReadFile,
```
Comment (ialidzhikov) [inline test/integration/shoots/operations/containerruntime.go:314]: These are the default values for `Resources`, `TerminationMessagePath` and  `TerminationMessagePolicy `, right? You could remove them.

**Comment by danielfoehrKn** [comment]
Decided that this integration test rather belongs [in the respective extension repo](https://github.com/gardener/gardener-extension-runtime-gvisor).

**Comment by danielfoehrKn** [comment]
Opened in the [gVisor-extension.](https://github.com/gardener/gardener-extension-runtime-gvisor/pull/6)

**Pull request closed without merging**